iot-hub/Tutorials/FirmwareUpdate/SimulatedDevice.js

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var Client = require('azure-iot-device').Client;
+var Protocol = require('azure-iot-device-mqtt').Mqtt;
+var async = require('async');
+const chalk = require('chalk');
+
+// Receive the IoT Hub device connection string as a command line parameter
+if(process.argv.length < 3) {
+  console.error('Usage: node SimulatedDevice.js <<IoT Hub Device Connection String>>');
+  process.exit(1);
+}
+
+var connectionString = process.argv[2];
+var client = Client.fromConnectionString(connectionString, Protocol);
+var desiredFirmwareProperties = null;
+var fwUpdateInProgress = false;
+var deviceTwin = null;
+
+// <reportedProperties>
+// Firmware update patch
+//  currentFwVersion: The firmware version currently running on the device.
+//  pendingFwVersion: The next version to update to, should match what's
+//                    specified in the desired properties. Blank if there
+//                    is no pending update (fwUpdateStatus is 'current').
+//  fwUpdateStatus:   Defines the progress of the update so that it can be
+//                    categorized from a summary view. One of:
+//     - current:     There is no pending firmware update. currentFwVersion should
+//                    match fwVersion from desired properties.
+//     - downloading: Firmware update image is downloading.
+//     - verifying:   Verifying image file checksum and any other validations.
+//     - applying:    Update to the new image file is in progress.
+//     - rebooting:   Device is rebooting as part of update process.
+//     - error:       An error occurred during the update process. Additional details
+//                    should be specified in fwUpdateSubstatus.
+//     - rolledback:  Update rolled back to the previous version due to an error.
+//  fwUpdateSubstatus: Any additional detail for the fwUpdateStatus . May include
+//                     reasons for error or rollback states, or download %.
+//
+// var twinPatchFirmwareUpdate = {
+//   firmware: {
+//     currentFwVersion: '1.0.0',
+//     pendingFwVersion: '',
+//     fwUpdateStatus: 'current',
+//     fwUpdateSubstatus: '',
+//     lastFwUpdateStartTime: '',
+//     lastFwUpdateEndTime: ''
+//   }
+// };
+// </reportedProperties>
+
+// Send reported properties
+function sendReportedProperties(patch) {
```

Comment:
I don't think it's a great practice to not have a callback on an async op and use it as "fire and forget".. what's the reason?